enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,597 @@
+
+Updating an existing IngressController with new subnets requires the service
+to be deleted and recreated:
+
+1. A cluster admin edits an IngressController's
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` field.
+2. The Ingress Operator sets the `LoadBalancerProgressing` condition to `Status: True`
+   on the IngressController and provides a message indicating to the cluster admin
+   that they must delete the service.
+3. The cluster admin deletes the LoadBalancer-type service indicated in the
+   `LoadBalancerProgressing` condition message.
+4. The Ingress Operator recreates the LoadBalancer-type service and configures the
+   `service.beta.kubernetes.io/aws-load-balancer-subnets` on the LoadBalancer-type
+   service, thus creating the load balancer with the desired subnets specified by the
+   cluster admin.
+
+#### Unmanaged Subnet Annotation Migration Workflow
+
+Migrating an unmanaged `service.beta.kubernetes.io/aws-load-balancer-subnets`
+service annotation to `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+after upgrading to 4.y doesn't require a cluster admin to delete the LoadBalancer-type
+service:
+
+1. Cluster admin confirms that initially `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   is set on the LoadBalancer-type service managed by the IngressController.
+2. The cluster admin upgrades OpenShift to v4.y and the service annotation is not
+   changed or removed.
+3. After upgrading, the IngressController will emit a `LoadBalancerProgressing` condition
+   with `Status: True` because the spec's `Subnets` (an empty slice `[]`) does not equal
+   the current annotation.
+4. Instead of recreating the LoadBalancer-type service, the cluster admin will set
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` to the
+   current value of the `service.beta.kubernetes.io/aws-load-balancer-subnets` service.
+5. The Ingress Operator will resolve the `LoadBalancerProgressing` condition back to
+   `Status: False` as long as `Subnets` is equivalent to `service.beta.kubernetes.io/aws-load-balancer-subnets`.
+
+### API Extensions
+
+This proposal will modify the `IngressController` API by adding a new
+field called `Subnets` of type `[]string` to the `AWSLoadBalancerParameters`
+struct type.
+
+TODO: What will install-time option modify?
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This enhancement is directly enabling the https://issues.redhat.com/browse/XCMSTRAT-545
+feature for ROSA Hosted Control Planes.
+
+#### Standalone Clusters
+
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+When an IngressController is created with
```

Comment:
It would be better to see this entire section earlier in the doc instead of down here.  The workflows would have made better sense.

Reply:
I'm just using the template: https://github.com/openshift/enhancements/blob/da6500436b19e70f44834aff1b01f578b130862f/guidelines/enhancement_template.md

But I agree. I can move it up, but I know that will be painful to review if you like reviewing the GitHub deltas, so I'll do it in a separate push.